### DISCUSSION TITLE
how to create travel guide?

### DISCUSSION THREAD

**Discussion by mariush444** [osmandapp/OsmAnd#12297] [Q&A]
I tried to follow by https://osmand.net/features/travel but no success.

I have two files `njz.gpx (223kB) and njz.html (123kB)` in directory `njz`
then I run command `./utilities.sh travel-guide-creator` for directory `njz`
The file `travel_guide.sqlite (143kB)` was created.

The sqlite file was put to net.osmand.plus/files/travel directory.

After restarting osmand I can't find any new travel guide in travel menu.

Does anyone has an idea?

[njz.zip](https://github.com/osmandapp/OsmAnd/files/6896138/njz.zip)

**Comment by vshcherb** [comment]
Hi Mars situation a little bit changed and we are going to update docs. Now Travel Guide is OBF file.  So if you have sqlite file created : 
- you need to process it further to OSM  ```OsmAndMapCreator/utilities.sh process-wikivoyage "$SQLITE" --generate-osm="$LOC/wikivoyage.osm.gz"```
- Convert osm.gz using MapCreator further to OBF ```OsmAndMapCreator/utilities.sh generate-poi default_wikivoyage.osm --chars-build-poi-nameindex=3```
-  later put in the folder ```osmand/travel/```

You can double check created OBF with binaryInspector

**Reply by mariush444** [reply]
@vshcherb   Thank you.
I got error running the first command:

```
Jul 29, 2021 4:49:05 PM net.osmand.travel.WikivoyageGenOSM getDatabaseConnection
INFO: SQLITE running in native mode
Exception in thread "main" java.sql.SQLException: [SQLITE_ERROR] SQL error or missing database (no such column: is_parent_of)
	at org.sqlite.DB.newSQLException(DB.java:383)
	at org.sqlite.DB.newSQLException(DB.java:387)
	at org.sqlite.DB.throwex(DB.java:374)
	at org.sqlite.NativeDB.prepare(Native Method)
	at org.sqlite.DB.prepare(DB.java:123)
	at org.sqlite.Stmt.executeQuery(Stmt.java:121)
	at net.osmand.travel.WikivoyageGenOSM.genWikivoyageOsm(WikivoyageGenOSM.java:156)
	at net.osmand.MainUtilities.main(MainUtilities.java:141)
```
The structure of the sqlite file
![image](https://user-images.githubusercontent.com/66887280/127528412-8d60f946-db8d-43e9-b1db-695e7437c79f.png)

I wonder whether sqlite file is OK because it's smaller than gpx file.

**Comment by mariush444** [comment]
I think that problem is during generating sqlite file. The structure is wrong.
I just add column `is_parent_of` to table `travel_articles` and this error disappear but I got new one
```
lip 29, 2021 9:16:28 PM net.osmand.travel.WikivoyageGenOSM getDatabaseConnection
INFO: SQLITE running in native mode
Exception in thread "main" java.io.EOFException
        at java.base/java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:268)
        at java.base/java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:259)
        at java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:164)
        at java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79)
        at java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:91)
        at net.osmand.travel.WikivoyageGenOSM.genWikivoyageOsm(WikivoyageGenOSM.java:192)
        at net.osmand.MainUtilities.main(MainUtilities.java:141)

```

**Reply by vshcherb** [reply]
The structure  / missing columns could be added via
```
OsmAndMapCreator/utilities.sh process-wikivoyage "$FILE"
```
but it's not neccessary it still important to run
```
OsmAndMapCreator/utilities.sh process-wikivoyage "$SQLITE" --generate-osm="$FILE.osm.gz"```
```
It's different command!

This is an sql and columns needed to generate OSM -> OBF
```
ResultSet rs = statement.executeQuery("select trip_id, title, lang, lat, lon, content_gz, "
				+ "gpx_gz, image_title, banner_title, is_part_of, is_parent_of, aggregated_part_of, contents_json from travel_articles order by trip_id asc");
```

**Comment by mariush444** [comment]
![image](https://user-images.githubusercontent.com/66887280/127661966-760ddf0c-3faf-4f21-918c-b60a1f03309b.png)

After creating sqlite file I see data only in content_gz and banner_tile.
gpz_gz is empty or there are wrong data. It seems that data are shifted according to columns.
`gpz_gz=1` but `1` should be in `trip_id`
`trip_id=1` but this `1` should be in `original_id`
`original_id=en` but `en` should be in `lang`.

I think that problem is at the first step` ./utilites.sh travel-guide-creator`

**Reply by vshcherb** [reply]
Well yes we need to fix it ```./utilites.sh travel-guide-creator``` but now we are in 4.0 release.

Created to fix later on https://github.com/osmandapp/OsmAnd/issues/12324

**Reply by vshcherb** [reply]
Hello it should work now !
./utilites.sh travel-guide-creator DIRECTORY_PATH

Will produce XXXX.travel.obf which you need to put /travel folder and it should work.

![telegram-cloud-photo-size-2-5240339488645952973-y](https://user-images.githubusercontent.com/1042025/129365655-87459061-d5ea-4bef-8513-b37887e056c5.jpg)
![telegram-cloud-photo-size-2-5240339488645952974-y](https://user-images.githubusercontent.com/1042025/129365676-408e057a-64e6-401a-91da-1cfcc98ea5cd.jpg)

**Reply by mariush444** [reply]
Thank you.
the obf file is indeed created in one step/command. Great.

But I found some problems
e.g. 
3.9.10 - I can read the guide but waypoints are in wrong places in the map and the track is not shown at all.
After some clicks osmand crushes.
4.0.5 - I can't find guide in Travel guide option but I see it in the list of downloaded wiki-guides.

I'll investigate this a little bit more then I create new discussion or issue unless you prefer to continue the topic in this post.
What are your preferences?
